src/components/Calendar.tsx

```diff
@@ -69,10 +73,36 @@ const getMarkedDates = (data?: any[]) => {
 
 export const Calendar = ({ query, queryVariables, calendarData, isLoading, navigation }: Props) => {
   const contentContainerId = queryVariables?.contentContainerId;
+  const today = moment().format('YYYY-MM-DD');
+  const { globalSettings } = useContext(SettingsContext);
+  const { settings = {} } = globalSettings;
+  const { eventCalendar = {} } = settings;
+  const { maxDotPerDay: MAX_DOTS_PER_DAY = 5, subList = false } = eventCalendar;
+
+  const [queryVariableWithDateRange, setQueryVariableWithDateRange] = useState<any>({
+    queryVariables,
+    dateRange: [today, today]
+  });
+  const [markedDates, setMarkedDates] = useState<CalendarProps['markedDates']>({});
+
+  const { data, loading } = useQuery(getQuery(QUERY_TYPES.EVENT_RECORDS), {
+    variables: queryVariableWithDateRange
+  });
+
+  useEffect(() => {
+    setMarkedDates(getMarkedDates(calendarData, MAX_DOTS_PER_DAY, today));
+  }, []);
 
   const onDayPress = useCallback(
     (day: DateData) => {
-      if (query === QUERY_TYPES.EVENT_RECORDS) {
+      if (query === QUERY_TYPES.EVENT_RECORDS && subList) {
+        setQueryVariableWithDateRange({
+          ...queryVariables,
+          dateRange: [day.dateString, day.dateString]
+        });
+        setMarkedDates(getMarkedDates(calendarData, MAX_DOTS_PER_DAY, day.dateString));
+        return;
+      } else if (query === QUERY_TYPES.EVENT_RECORDS) {
```

Comment:
to have less changes in code you could write it this way:

```
if (query === QUERY_TYPES.EVENT_RECORDS) {
  if (subList) {
    setQueryVariableWithDateRange({
      ...queryVariables,
      dateRange: [day.dateString, day.dateString]
    });
    setMarkedDates(getMarkedDates(calendarData, MAX_DOTS_PER_DAY, day.dateString));

    return;
  }

  return navigation.push(ScreenName.Index, {
    title: texts.homeTitles.events,
    query,
    queryVariables: { ...queryVariables, dateRange: [day.dateString, day.dateString] },
    rootRouteName: ROOT_ROUTE_NAMES.EVENT_RECORDS
  });
}
```

or

```
if (query === QUERY_TYPES.EVENT_RECORDS) {
  if (subList) {
    setQueryVariableWithDateRange({
      ...queryVariables,
      dateRange: [day.dateString, day.dateString]
    });
    setMarkedDates(getMarkedDates(calendarData, MAX_DOTS_PER_DAY, day.dateString));
  } else {
    return navigation.push(ScreenName.Index, {
      title: texts.homeTitles.events,
      query,
      queryVariables: { ...queryVariables, dateRange: [day.dateString, day.dateString] },
      rootRouteName: ROOT_ROUTE_NAMES.EVENT_RECORDS
    });
  }
}
```

what is your preference and why?